client/rest-high-level/src/main/java/org/elasticsearch/client/ml/PostDataRequest.java

```
@@ -0,0 +1,224 @@
+     *
+     * @throws IOException on parsing/serialization errors
+     */
+    public BytesReference getContent() throws IOException {
+        if (content != null) {
+            return content;
+        }
+        try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
+            for (BytesReference bytesReference : bytesReferences) {
+                try (StreamInput streamInput = bytesReference.streamInput()) {
+                    builder.rawValue(streamInput, xContentType);
+                }
+            }
+            for (Map<String, Object> objectMap : objectMaps) {
+                builder.map(objectMap);
+            }
+            return BytesReference.bytes(builder);
+        }
+    }
+
+    public XContentType getXContentType() {
+        return xContentType;
+    }
+
+    /**
+     * Set the total content to post.
+     *
+     * @param content BytesReference content to set, format must match the set XContentType
+     */
+    public void setContent(BytesReference content) {
+        this.content = content;
+    }
+
+    /**
+     * Add a document via a ByteReference.
+     *
+     * Ignored if total content is set via {@link PostDataRequest#setContent(BytesReference)}
+     *
+     * @param bytesReference document to add to bulk request, format must match the set XContentType
+     */
+    public void addDoc(BytesReference bytesReference) {
+        this.bytesReferences.add(Objects.requireNonNull(bytesReference, "bytesReferences must not be null"));
+    }
+
+    /**
+     * Add a document via an object map
+     *
+     * Ignored if total content is set via {@link PostDataRequest#setContent(BytesReference)}
+     *
+     * @param objectMap document object to add to bulk request
+     */
+    public void addDoc(Map<String, Object> objectMap) {
+        this.objectMaps.add(Objects.requireNonNull(objectMap, "objectMap must not be null"));
```

Should we throw an exception if `content != null`?  Or otherwise `return` early to ignore as the Javadoc says.